### ISSUE TITLE
Regex rules with exists:0 fails

### ISSUE THREAD

**Issue by ghostrider-05** [loeffel-io/ls-lint#243]
I want to add a rule that does not allow files to be named `image\d+` and [according to the documentation](https://ls-lint.org/2.3/configuration/the-rules.html#exists) `exists:0` can be used as:

```yml
ls:
   .png: regex:image\d+ | exists:0
``` 

However, when running with this configuration all the files that are named `image\d+` are passing and files not named `image\d+` are failing with:

```
custom_image.png failed for `.png` rules: regex:image\d+
```

I thought `exists:0` would be applied as a seperate rule, so I removed the `|` and tried again with the following configuration:

```yml
ls:
   .png: regex:image\d+ exists:0
``` 
With both images failing the test:

```
custom_image.png failed for `.png` rules: regex:image\d+ exists:0
image214.png failed for `.png` rules: regex:image\d+ exists:0
```

Is this intended behaviour? In the documentation I don't see an example with `exists:N` and regex rules so I thought it should be possible.

Version: `2.3.0-beta.1` on NPM

**Comment by loeffel-io** [comment]
Hey @ghostrider-05, thank you for your contribution!

this is intended.

The `exists` rule is exclusive - `exists:0` just make sure that there is no file with this extension while it can make sense to combine `exists:1` or `exists:N-M` with other rules.

i think you need something like a `not_regex` rule right?

**Comment by ghostrider-05** [comment]
Ah thank you for the clarification, sorry if this has been asked before. A `not_regex` rule would be perfect since regex lookahead is not possible

**Comment by ccoVeille** [comment]
Well I would expect to have lookahead regexp instead of not_regexp.

Would it be possible?

**Comment by ccoVeille** [comment]
> this is intended.\n\nThe exists rule is exclusive - exists:0 just make sure that there is no file with this extension while it can make sense to combine exists:1 or exists:1-M with other rules.

It could be added to the documentation then 🤭😅😬

**Comment by loeffel-io** [comment]
> Well I would expect to have lookahead regexp instead of not_regexp.
> 
> Would it be possible?

this is not possible because golang do not support lookaheads

**Comment by ccoVeille** [comment]
Go standard regexp doesn't support it.

But you could use another lib

https://github.com/dlclark/regexp2

**Comment by loeffel-io** [comment]
yeah but i would prefer `not_regex` which is pretty clear instead of a third party lib with complex lookaheads - wdyt @ccoVeille

**Comment by ccoVeille** [comment]
👍. But then the reason is what gave (and I'm fine with it) and not "go doesn't support lookahead"

**Comment by loeffel-io** [comment]
`not_regex` is now available in the v2.3.0-beta.2 release
Please test it and let me know if there are any issues ❤️
thanks to @RobinIrmer 

The upcoming v2.3.0 blog post: https://ls-lint.org/blog/announcements/v2.3.0.html
The docs: https://ls-lint.org/2.3/getting-started/introduction.html
The release notes: https://github.com/loeffel-io/ls-lint/releases/tag/v2.3.0-beta.2

**Comment by loeffel-io** [comment]
There was an issue with `v2.3.0-beta.2` - `v2.3.0-beta.3` is the correct version.
sorry for any inconveniences

**Comment by ghostrider-05** [comment]
Had time to test it finally on `v2.3.0-beta.3` and it is working as expected!

**Comment by loeffel-io** [comment]
> Had time to test it finally on `v2.3.0-beta.3` and it is working as expected!

great! maybe i will change the syntax a bit 
from not_regex:[a-z]+ to regex:![a-z]+ to keep anything inside the regex rule but i will think about it

**Comment by loeffel-io** [comment]
`v2.3.0` is finally here: https://ls-lint.org/blog/announcements/v2.3.0.html
